Sources/Commands/SwiftPackageCollectionsTool.swift

```diff
@@ -0,0 +1,358 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright 2020 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import ArgumentParser
+import Foundation
+import PackageCollections
+import PackageModel
+import TSCBasic
+import TSCUtility
+
+private enum CollectionsError: Swift.Error {
+    case invalidVersionString(String)
+    case missingArgument(String)
+    case noCollectionMatchingURL(String)
+}
+
+extension CollectionsError: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .invalidVersionString(let versionString):
+            return "invalid version string '\(versionString)'"
+        case .missingArgument(let argumentName):
+            return "missing argument '\(argumentName)'"
+        case .noCollectionMatchingURL(let url):
+            return "no collection matching URL '\(url)'"
+        }
+    }
+}
+
+struct JSONOptions: ParsableArguments {
+    @Flag(name: .long, help: "Output as JSON")
+    var json: Bool = false
+}
+
+struct ProfileOptions: ParsableArguments {
+    @Option(name: .long, help: "Profile to use for the given command")
+    var profile: String?
+
+    var usedProfile: PackageCollectionsModel.Profile {
+        if let profile = profile {
+            return .init(name: profile)
+        } else {
+            return .default
+        }
+    }
+}
+
+public struct SwiftPackageCollectionsTool: ParsableCommand {
+    public static var configuration = CommandConfiguration(
+        commandName: "package-collections",
+        _superCommandName: "swift",
+        abstract: "Interact with package collections",
+        discussion: "SEE ALSO: swift build, swift package, swift run, swift test",
+        version: Versioning.currentVersion.completeDisplayString,
+        subcommands: [
+            Add.self,
+            Describe.self,
+            DescribeCollection.self,
+            List.self,
+            ProfileList.self,
+            Refresh.self,
+            Remove.self,
+            Search.self
+        ],
+        helpNames: [.short, .long, .customLong("help", withSingleDash: true)])
+
+    public init() {
+    }
+
+    // MARK: Profiles
+
+    struct ProfileList: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "List configured profiles")
+
+        @OptionGroup
+        var jsonOptions: JSONOptions
+
+        mutating func run() throws {
```

Comment:
Do all the `run()`s really need to be `mutating`?

Reply:
Yah, that's how this method is defined in the `ParsableCommand` protocol.